.travis.yml

```
@@ -2,24 +2,31 @@ language: cpp
 os: linux
 compiler: gcc
 
+cache:
+- directories:
+  - $HOME/.cache
+
 addons:
   apt:
     packages: lcov
 
-before_script:
-- mkdir -p build && cd build
+install:
+# (fake) install dependencies
+- mkdir -p $HOME/.cache
+- touch $HOME/.cache/mydepency.cpp
 
-script:
-- cmake -DCODE_COVERAGE=ON ..
-- cmake --build . --config Debug -- -j $(nproc)
-- ctest -j $(nproc) --output-on-failure
+script: ./build.sh
 
 after_success:
 # Create lcov report
-- lcov --capture --directory . --output-file coverage.info
-- lcov --remove coverage.info '/usr/*' --output-file coverage.info # filter system-files
+# capture coverage info
+- lcov --directory . --capture --output-file coverage.info
+# filter out system and extra files
+- lcov --remove coverage.info '/usr/*' '*/tests/*' "${HOME}"'/.cache/*' --output-file coverage.info
```

```suggestion
- lcov --remove coverage.info '/usr/*' "${HOME}"'/.cache/*' --output-file coverage.info
```
It is not uncommon to leave the coverage of the tests in.
In this simple example it does not add any information, but for more complicated tests it gives insight into test execution. Especially with CodeCov showing which build configuration executed the code. (And unit tests do tend to contain many configuration specific parts.)

---

Agreed. However this is intended as an example how to **not** include test coverage. Again the README.MD shows basic usage and I added a part about excluding test code coverage which is showed in the "full" implementation. This code is obviously intended to be customized to suit own needs and removing things one does not need is easier than adding them. I e.g. tried `tests/*` as the pattern but that does not work.

---

Hmm, but this way it looks as if you are advising to exclude the tests.
A comment saying a full path match is required with an example in the readme could attend on difficulties.
e.g. `# source folder names must match a full path, e.g.: '*/external/*`

Same for showing how to remove multiple matches in a single call. Nice to show, but it results in a build specific call. Besides, these are file-edits. Even when someone is too lazy to check the lcov manual there is not much lost in repeated calls.

---

Better: 83e43da ?

On the last point: I could either show hot to do multi-matches with single call or that you can do multiple calls. As you could argue for/against either I'd just leave it like this.